docs/source/testing-and-debugging.md

```diff
@@ -43,71 +43,79 @@ from boa3.boa3 import Boa3
-function you'll need a NeoTestRunner object, which takes the path of your Neo-Express network configuration file as an
-argument to set up the test environment.
+Then, create functions to test the expected behavior of your smart contract. To invoke or test invoke your smart 
+contract, use the `call` method from `SmartContractTestCase`. The two positional parameters are the name of the method 
+you want to invoke and a list of its arguments. The keyword parameters are the return type, a list of signing accounts,
+a list of signers, and the smart contract you want to invoke. If you get an error when calling a smart contract, then an
+error will be raised.
 
-You'll have to call the method `call_contract()` to interact with your smart contract. Its parameters are the path of
-the compiled smart contract, the smart contract's method, and the arguments if necessary. 
-This call doesn't return the result directly, but includes it in a queue of invocations. To execute all the invocations
-set up, call the method `execute()`. Then assert the result of your invoke to see if it's correct.
-
-Note that `invoke.result` won't be set if the execution fails, so you should also assert if `runner.vm_state` is valid 
-for your test case.
+To persist an invocation, use the `signing_accounts` parameter to pass a list of signing accounts when calling the 
+smart contract. If you don't pass it, then it will always be a test invoke. The `signers` parameter can be used 
+alongside the `signing_accounts` if you want to change the witness scope of the invocation, or by itself if you want to
+test invoke but also define the signers of the transaction.
 
 Your Python Script should look something like this:
 
 ```python
-
-from boa3.sc.types import VMState
-from boa3_test.test_drive.testrunner.neo_test_runner import NeoTestRunner
-
-
-def test_hello_world_main():
-    neoxp_config_file = '{path-to-neo-express-config-file}'
-    project_root_folder = '{path-to-project-root-folder}'
-    path = f'{project_root_folder}/boa3_test/examples/hello_world.nef'
-    runner = NeoTestRunner(neoxp_config_file)
-
-    invoke = runner.call_contract(path, 'main')
-    runner.execute()
-    assert runner.vm_state is VMState.HALT
-    assert invoke.result is None
+import asyncio
+from boaconstructor import SmartContractTestCase
+from neo3.core import types
+from neo3.wallet.account import Account
+from neo3.contracts.contract import CONTRACT_HASHES
+
+GAS = CONTRACT_HASHES.GAS_TOKEN
+
+class SmartContractTest(SmartContractTestCase):
+    genesis: Account
+    user1: Account
+    
+    # if this variable is set, then this contract hash will be called whenever you don't use the `target_contract` parameter on the `call` method
+    contract_hash: types.UInt160
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        # you can name the account whatever you want, but the password needs to be "123"
```

Comment:
is there a reason that the password needs to be "123"? is it a boa-test-case issue?